Src/FluentAssertions/Events/EventAssertions.cs

```diff
@@ -91,18 +91,31 @@ public void NotRaise(string eventName, string because = "", params object[] beca
         public IEventRecording RaisePropertyChangeFor(Expression<Func<T, object>> propertyExpression,
             string because = "", params object[] becauseArgs)
         {
-            IEventRecording recording = monitor.GetRecordingFor(PropertyChangedEventName);
             string propertyName = propertyExpression?.GetPropertyInfo().Name;
 
+            IEventRecording recording = monitor.GetRecordingFor(PropertyChangedEventName);
+
             if (!recording.Any())
             {
                 Execute.Assertion
                     .BecauseOf(because, becauseArgs)
-                    .FailWith("Expected object {0} to raise event {1} for property {2}{reason}, but it did not.",
+                    .FailWith("Expected object {0} to raise event {1} for property {2}{reason}, but it did not raise that event at all.",
                         monitor.Subject, PropertyChangedEventName, propertyName);
             }
 
-            return recording.WithArgs<PropertyChangedEventArgs>(args => args.PropertyName == propertyName);
+            var actualPropertyNames = recording
+                .SelectMany(@event => @event.Parameters.OfType<PropertyChangedEventArgs>())
+                .Select(eventArgs => eventArgs.PropertyName)
```

Comment:
Shouldn't it be singular?

Reply:
Nope. The object itself is named like that.

Reply:
Aha, finally it makes sense to me